tests/test_job_tracker_v2.py

```diff
@@ -751,9 +927,65 @@ def test_parse_application_report_running(self):
             "memory": {"unit": "mb-seconds", "value": 96183879},
         }
 
-    def test_parse_application_report_empty(self):
+    def test_parse_application_response_empty(self):
         with pytest.raises(YarnAppReportParseException):
-            _ = YarnStatusGetter().parse_application_report(report="")
+            _ = YarnStatusGetter.parse_application_response(json={})
+
+    def test_get_authentication_provider_use_kerberos_on(self, monkeypatch):
+        """Simple check that the config has the desired effect: we get the expected auth object."""
+        monkeypatch.setenv("YARN_AUTH_USE_KERBEROS", "yes")
+        auth = YarnStatusGetter().get_authentication_provider()
+        assert isinstance(auth, requests_gssapi.HTTPKerberosAuth)
+
+    def test_get_authentication_provider_use_kerberos_off(self, monkeypatch):
+        """Simple check that the config has the desired effect: authentication should be None."""
+        monkeypatch.setenv("YARN_AUTH_USE_KERBEROS", "no")
+        assert YarnStatusGetter().get_authentication_provider() is None
+
+    def test_kerberos_auth_should_fail_when_there_is_no_kerberos_setup(
```

Comment:
I don't think this is a good name because this is not the behavior that you're trying to assert, it's just the way it's implemented.

Maybe `test_kerberos_auth_is_actually_attempted()` is better (like the docstring describes).

Reply:
I ended up removing this test too.
After the refactoring it was not a very useful test anymore because there was only one place left where we need to instantiate the Kerberos authentication.